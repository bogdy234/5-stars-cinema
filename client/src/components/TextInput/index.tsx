import { FC, ReactElement, useState } from "react";
import Icon from "../Icon";

interface TextInputProps {
    placeholder?: string;
    value: string;
    onChange: (newValue: string) => void;
    label?: string;
    className?: string;
    inputClassName?: string;
    labelClassName?: string;
    password?: boolean;
}

const TextInput: FC<TextInputProps> = ({
    placeholder,
    value,
    onChange,
    label,
    className,
    inputClassName,
    labelClassName,
    password,
}: TextInputProps): ReactElement => {
    const [showPassword, setShowPassword] = useState<boolean>(
        password || false
    );

    const onChangeInputType = () => {
        setShowPassword((a) => !a);
    };

    return (
        <div className={className}>
            <label className={`${labelClassName}`}>
                {label}
                <div className={`flex items-center relative max-w-sm`}>
                    <input
                        type={showPassword ? "password" : "text"}
                        placeholder={placeholder}
                        value={value}
                        onChange={(e) => onChange(e.target.value)}
                        className={`w-full h-10 pl-4 pr-10 ${inputClassName}`}
                    />
                    {password && (
                        <Icon
                            src="eye-solid.svg"
                            alt="eye"
                            className={`w-4 absolute right-0`}
                            onClick={onChangeInputType}
                        />
                    )}
                </div>
            </label>
        </div>
    );
};

export default TextInput;
