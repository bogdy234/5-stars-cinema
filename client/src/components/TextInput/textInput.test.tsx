import renderer from "react-test-renderer";
import TextInput from "./index";

it("renders correctly", () => {
  const tree = renderer
    .create(<TextInput value={""} onChange={jest.fn()} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
