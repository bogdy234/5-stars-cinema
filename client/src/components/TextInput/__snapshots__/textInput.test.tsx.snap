// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className=""
>
  <label
    className="undefined || ''"
  >
    <div
      className="flex items-center relative undefined"
    >
      <input
        className="w-full h-10 pl-4 pr-10 undefined undefined"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
  </label>
</div>
`;
